api/odigos/v1alpha1/source_types.go

```diff
@@ -79,6 +80,14 @@ type SourceSpec struct {
 	// +kubebuilder:validation:Optional
 	// +optional
 	ContainerOverrides []ContainerOverride `json:"containerOverrides,omitempty"`
+
+	// UseRegex indicates that the workload name should be matched using regex.
+	// When true, spec.workload.name is treated as a regular expression pattern.
+	// This allows matching multiple workloads with a single Source CRD.
+	// Not valid for namespace sources.
+	// +kubebuilder:validation:Optional
+	// +optional
+	UseRegex bool `json:"useRegex,omitempty"`
```

Comment:
WDYT about naming it

```suggestion
	MatchWorkloadNameAsRegex bool `json:"matchWorkloadNameAsRegex,omitempty"`
```

The comment is helpful, but if someone just look at one specific Source object using kubectl, they will see `useRegex` which can be cryptic without the right context.